build-system/tasks/dep-check.js

```diff
@@ -204,9 +203,12 @@ function getGraph(entryModule) {
  */
 function getEntryModule(extensionFolder) {
   var extension = path.basename(extensionFolder);
-  var versions = extensionsVersions[extension] || ['0.1'];
-  return versions.map(v =>
-      `${extensionFolder}/${v}/${extension}.js`);
+  return fs.readdirAsync(extensionFolder).then(dirItems => {
```

Comment:
As a personal opinion I'm not keen on adding `_depr` to the folder name since folders are supposed to be version numbers. The extension is marked deprecated in the protoascii which warns publishers. Perhaps an alternative to renaming a folder would be to update the `.js` file itself with a deprecated heading or adding a file named `DEPRECATED` in the versioned directory.

As to `dep_check`, since changes could still be made to these files (AFAIK nothing prevents that other than watchful eyes) and also it may be necessary to make changes for a fix or security issue, I'd think we would still want to keep the checks since they're live files. That said these are build items and I am nowhere near understanding all that you and other awesome releasers do to make things work.